strawberryfields/gbs/vibronic.py

```diff
@@ -0,0 +1,228 @@
+
+.. math::
+    U_{Dok} = C_L \Sigma C_R^t D_\alpha.
+
+The squeezing and rotation operators are obtained by singular value decomposition of a matrix
+:math:`J` as
+
+.. math::
+    J = C_L \Sigma C_R^t.
+
+In order to obtain :math:`J` we need to construct diagonal matrices :math:`\Omega` and
+:math:`\Omega'` from square roots of the ground state and excited state normal mode frequencies,
+:math:`\omega` and :math:`\omega'`, respectively. Then :math:`J` is obtained as
+
+.. math::
+    J = \Omega' U \Omega^{-1},
+
+where
+
+.. math::
+    \Omega = diag (\sqrt{\omega_1},...,\sqrt{\omega_k})
+
+    \Omega' = diag (\sqrt{\omega_1'},...,\sqrt{\omega_k'})
+
+and :math:`U` is a molecular coordinate transformation known as the Duschinsky matrix which is
+obtained from the normal mode coordinates of the ground and excited states, :math:`q` and
+:math:`q'` respectively, as
+
+.. math::
+    q' = Uq + d.
+
+The displacement parameter :math:`d` is related to the structural changes of a molecule upon
+vibronic excitation and is related to the displacement parameter :math:`\alpha` as
+
+.. math::
+    \alpha = d / \sqrt{2}.
+
+In general, all the GBS parameters can be derived from the previously computed equilibrium
+geometries, vibrational frequencies and normal coordinates of the molecule in its (electronic)
+ground and excited states.
+
+
+Summary
+-------
+
+.. autosummary::
+    gbs_params
+    energy
+    spectrum
+
+Code details
+^^^^^^^^^^^^
+"""
+from typing import Tuple
+
+import numpy as np
+
+
+def gbs_params(
+    w: np.ndarray, wp: np.ndarray, Ud: np.ndarray, d: np.ndarray
+) -> Tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray]:
+    r"""Converts molecular information input to Gaussian gate parameters.
+
+    We first construct diagonal matrices :math:`\Omega` and :math:`\Omega'` from square roots of
+    the ground state and excited state normal mode frequencies, :math:`\omega` and :math:`\omega'`,
+    respectively. Then we construct the matrix :math:`J` as
+
+    .. math::
+        J = \Omega' U \Omega^{-1},
+
+    where
+
+    .. math::
+        \Omega = diag (\sqrt{\omega_1},...,\sqrt{\omega_k}),
+
+        \Omega' = diag (\sqrt{\omega_1'},...,\sqrt{\omega_k'})
+
+    and :math:`U` is a molecular coordinate transformation known as the Duschinsky matrix.
+    The squeezing and rotation operators are obtained by singular value decomposition of the matrix
+    :math:`J` as
+
+    .. math::
+        J = C_L \Sigma C_R^t.
+
+    The displacement parameter :math:`\alpha` is obtained from the Duschinsky displacement
+    parameter :math:`d` as:
+
+    .. math::
+        \alpha = d / \sqrt{2}.
+
+
+    **Example usage:**
+
+    >>> w = np.array([3765.2386, 3088.1826, 1825.1799, 1416.9512, 1326.4684, 1137.0490, 629.7144])
+    >>> wp = np.array([3629.9472, 3064.9143, 1566.4602, 1399.6554, 1215.3421, 1190.9077, 496.2845])
+    >>> delta = np.array([0.2254, 0.1469, 1.5599, -0.3784, 0.4553, -0.3439, 0.0618])
```

Comment:
Would be good for the variables to match the argument names, e.g., `U` could be `Ud`